src/index.tsx

```diff
@@ -22,16 +22,16 @@ export const createService = <A, P = {}>(
 }
 
 const createServiceComponent = <A, P = {}>(
-  context: React.Context<A>,
-  useApi: (props: Omit<P, 'children' | 'fallback'>) => A
+  context: React.Context<A | undefined>,
+  useApi: (props: P) => A
 ): ServiceComponent<A, P> => {
   const { Provider } = context
   const Service: ServiceComponent<A, P> = ({
     children,
     fallback: Fallback,
     ...props
   }) => {
-    const api = useApi(props)
+    const api = useApi(props as unknown as P)
```

Comment:
Necessary evil I believe, unless you can work it out. Basically it expects props to contain children and fallback, even though I don't think it should.